### PR TITLE
gluon-client-bridge: add WPA3 Enhanced Open client radio support

### PR DESCRIPTION
**Pull request by blocktrron**
This PR adds support for an additional WPA3 Enhanced Open Clients radio.

Enhanced Open provides an encrypted connection, leading to data transferred between the node and client device being protected against passive packet sniffing. It also makes use of mandatory MFP, eliminating deauthentication attacks.

Enahnced Open is fully supported on Android 10, so adoption will most likely increase in the next 1-3 years. 

On the AP side, OWE is (due to enforced MFP) supported on the following chipsets:

 * ath9k (exceptions apply for early draft-N chips)
 * ath10k
 * mt76

Blocked by #1876

### CODE REVIEW

File: docs/dev/hardware.rst [hunk 1]
```diff
@@ -9,6 +9,16 @@ Having an ath9k, ath10k or mt76 based WLAN adapter is highly recommended,
 although other chipsets may also work. VAP (multiple SSID) support
 is a requirement.
 
+Device classes
+--------------
+Gluon is currently aware of two device classes. Dpending on the device class, different
```
Comment (mweinelt) [inline docs/dev/hardware.rst]:
```suggestion
Gluon is currently aware of two device classes. Depending on the device class, different
```

File: docs/dev/hardware.rst [hunk 2]
```diff
@@ -9,6 +9,16 @@ Having an ath9k, ath10k or mt76 based WLAN adapter is highly recommended,
...
+features can be installed on the device.
```
Comment (mweinelt) [inline docs/dev/hardware.rst]:
```suggestion
features can be installed onto the device.
```

File: docs/site-example/site.conf [hunk 3]
```diff
@@ -48,6 +48,12 @@
       -- disabled = true, -- (optional)
     },
 
+    -- Configuration for OWE network.
```
Comment (mweinelt) [inline docs/site-example/site.conf]:
```suggestion
    -- Configuration for a non-backward compatible OWE network.
```

File: docs/site-example/site.conf [hunk 4]
```diff
@@ -65,6 +71,9 @@
     ap = {
       ssid = 'alpha-centauri.freifunk.net',
     },
+    owe = {
```
Comment (mweinelt) [inline docs/site-example/site.conf]: How about making this an example for a backwards compatible (`transition_mode = true,`) OWE network?

File: docs/user/site.rst [hunk 5]
```diff
@@ -124,6 +124,7 @@ wifi24 \: optional
     configure any subset of them:
 
     - ``ap`` creates a master interface where clients may connect
+    - ``owe`` creates a OWE interface where clients may connect (requires ``ap`` to be present)
```
Comment (mweinelt) [inline docs/user/site.rst]: Does it? Even if no transition mode was enabled?

File: docs/user/site.rst [hunk 6]
```diff
@@ -133,6 +134,12 @@ wifi24 \: optional
     ``ap`` requires a single parameter, a string, named ``ssid`` which sets the
     interface's ESSID. This is the WiFi the clients connect to.
 
+    ``owe`` requires a single parameter, a string, named ``ssid`` which sets the
+    interface's ESSID. This is the WiFi OWE capable clients can connect to.
+    Additionally, the ``transition_mode`` parameter enables the transition mode.
+    In this case, only the unencrypted SSID is visible. OWE capable clients will
+    automatically connect to the OWE network when selecting the unencrypted BSSID.
```
Comment (mweinelt) [inline docs/user/site.rst]:
```suggestion
    automatically connect to the OWE network based on information elements in the unencrypted SSIDs beacons.
```

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 7]
```diff
@@ -24,12 +23,9 @@ util.foreach_radio(uci, function(radio, index, config)
...
 	uci:delete('wireless', name)
 
-	if not ap() then
-		return
-	end
-
 	local macaddr = util.get_wlan_mac(uci, radio, index, 1)
-	if not macaddr then
+
+	if not ap() or not macaddr then
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: This prevents setting up a pure OWE network. Is that intentional?

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 8]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
 		ifname = suffix and 'client' .. suffix,
 		disabled = disabled or false,
 	})
+end
+
+local function configure_owe(radio, index, config, radio_name)
+	local name_client = 'client_' .. radio_name
+	local name_owe = 'owe_' .. radio_name
+	local suffix = radio_name:match('^radio(%d+)$')
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: radio_id?

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 9]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+	uci:section('wireless', 'wifi-iface', name_owe, {
+		device = radio_name,
+		network = 'client',
+		mode = 'ap',
+		ssid = owe.ssid(),
+		macaddr = macaddr,
+		ifname = suffix and 'owe' .. suffix,
+		disabled = disabled or false,
+		encryption = 'owe',
+		mfp = 2,
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: Shouldn't this vif be `hidden` if `transition_mode` was enabled?

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 10]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+	})
+end
+
+local function configure_owe_tm(radio, index, config, radio_name)
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]:
```suggestion
local function configure_owe_transition_mode(radio, index, config, radio_name)
```

We should probably spell it out.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 9 above)
Comment (mweinelt) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: Happens here https://github.com/freifunk-gluon/gluon/pull/1945/files#diff-09ec2a20205df7d6051ebd606cf29f0dR94.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 11]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+	uci:set('wireless', name_owe, 'owe_transition_bssid', macaddr_client)
+	uci:set('wireless', name_owe, 'hidden', '1')
+end
+
+util.foreach_radio(uci, function(radio, index, config)
+	local radio_name = radio['.name']
+
+	configure_ap(radio, index, config, radio_name)
+
+	if not platform.supports_wpa3() then
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: How is `supports_wpa3()` appropriate for feature gating OWE?

File: docs/user/site.rst (hunk 5 above)
Comment (blocktrron) [reply docs/user/site.rst]: Currently yes, as it's state is set implicitly from the state of the regular client network.

This is something which should be improved, but I'm not quite sure how to handle this in the web wizard, as the settings required to change is dependent whether or not we are using transition mode.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 7 above)
Comment (blocktrron) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: See my other comment - currently yes.

File: package/gluon-web-private-wifi/i18n/de.po [hunk 12]
```diff
@@ -13,18 +13,39 @@ msgstr ""
...
 msgstr "8-63 Zeichen"
 
+msgid "Disabled"
+msgstr "Deaktiviert"
+
 msgid "Enabled"
 msgstr "Aktiviert"
 
+msgid "Encryption"
+msgstr ""
```
Comment (mweinelt) [inline package/gluon-web-private-wifi/i18n/de.po]: Translations are missing here and below.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 8 above)
Comment (blocktrron) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: i sued to change to the old name (which is also fine IMHO)

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 9 above)
Comment (blocktrron) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: This is done in the tm_setup function.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 11 above)
Comment (blocktrron) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: OWe requires MFP (hardware feature) as well as hostapd-openssl (software-feature).

File: package/gluon-web-private-wifi/i18n/de.po (hunk 12 above)
Comment (blocktrron) [reply package/gluon-web-private-wifi/i18n/de.po]: Good catch. This is however something for the private-wpa3 PR. Will fix it over there.

File: package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua [hunk 13]
```diff
@@ -26,6 +27,25 @@ key:depends(enabled, true)
...
+	encryption:value("psk3", translate("WPA3"))
+end
+encryption.default = uci:get('wireless', primary_iface, 'encryption') or "psk2"
+
+local mfp = s:option(ListValue, "mfp", translate("Management Frame Protection"))
+mfp:depends(enabled, true)
+mfp:value("0", translate("Disabled"))
+if platform.supports_mfp(uci) then
+	mfp:value("1", translate("Optional"))
+	mfp:value("2", translate("Enabled"))
```
Comment (mweinelt) [inline package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua]: Required?

File: docs/site-example/site.conf (hunk 4 above)
Comment (blocktrron) [reply docs/site-example/site.conf]: The transition-mode is commented out in the 2.4GHz part, as it's an optional setting. These options do not seem to be included in the `wifi5` section, whus I've decided to skip it here.

File: package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua [hunk 14]
```diff
@@ -34,16 +54,24 @@ function f:write()
...
-				encryption = 'psk2',
+				encryption = encryption.data,
 				ssid       = ssid.data,
 				key        = key.data,
 				macaddr    = macaddr,
 				disabled   = false,
 			})
+
+			-- hostapd-mini won't start in case 802.11w is found in
+			-- the configuration
```
Comment (mweinelt) [inline package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua]: How do we address this?

Comment (blocktrron) [reply package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua]: The case directly below this commit solves this issue.

File: package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua (hunk 13 above)
Comment (blocktrron) [reply package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua]: This was the naming I've used initially, however MFP is not fully-enabled when It's only optional (and required is not the correct way to express such a condition), that's why I've decided to go with enabled here.

Comment (mweinelt) [reply package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua]: Can you rephrase that? I'm not understanding what you're saying.

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 11 above)
Comment (mweinelt) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: Maybe make that a comment.

File: package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua [hunk 15]
```diff
@@ -61,6 +61,10 @@ uci:foreach('wireless', 'wifi-device', function(config)
 
 	local p = f:section(Section, title)
 
+	local function has_owe_radio(radio)
```
Comment (mweinelt) [inline package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua]: > package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua:64:31: (W431) shadowing upvalue radio on line 51

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 16]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+		ssid = owe.ssid(),
+		macaddr = macaddr,
+		ifname = suffix and 'owe' .. suffix,
+		disabled = disabled or false,
+		encryption = 'owe',
+		mfp = 2,
+	})
+end
+
+local function configure_owe_transition_mode(radio, index, config, radio_name)
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: >    package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless:73:46: (W212) unused argument radio

>    package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless:73:53: (W212) unused argument index

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 17]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+
+	local ssid_client = uci:get('wireless', name_client, 'ssid')
+	local ssid_owe = uci:get('wireless', name_owe, 'ssid')
+
+	local macaddr_client = uci:get('wireless', name_client, 'macaddr')
+	local macaddr_owe = uci:get('wireless', name_owe, 'macaddr')
+
+	local owe = config.owe
+
+	if not owe() or not owe.transition_mode() or not ssid_client or not ssid_owe or not macaddr_client or not macaddr_owe then
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: >    package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless:85:121: (W631) line is too long (123 > 120)

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 18]
```diff
@@ -42,6 +38,73 @@ util.foreach_radio(uci, function(radio, index, config)
...
+		return
+	end
+
+	uci:set('wireless', name_client, 'owe_transition_ssid', ssid_owe)
+	uci:set('wireless', name_client, 'owe_transition_bssid', macaddr_owe)
+
```
Comment (mweinelt) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: >    package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless:91:1: (W611) line contains only whitespace

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless (hunk 11 above)
Comment (blocktrron) [reply package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: I've just double-checked - supports_wpa3 returns true in case a suitable hostapd is installed.

We acutally need to assert on supports_mfp here, as this implies the hostapd availability as well as all WiFi interfaces supporting MFP. This is not the case with broken draft-n Atheros chips.

**Comment by blocktrron** [comment]
Everything should be addressed, documentation still needs an overhaul to reflect the improved site structure.

**Comment by blocktrron** [comment]
Updated documentation to reflect the current implementation.

File: docs/dev/hardware.rst (hunk 1 above)
Comment (mweinelt) [inline docs/dev/hardware.rst]: I still see `Dpending` here.

File: docs/dev/hardware.rst [hunk 19]
```diff
@@ -9,6 +9,16 @@ Having an ath9k, ath10k or mt76 based WLAN adapter is highly recommended,
...
 
+Device classes
+--------------
+Gluon is currently aware of two device classes. Dpending on the device class, different
+features can be installed onto the device.
+
+The ``tiny`` device-class contains the following type of devices:
+
+* All devices with less than 64 MB of system memory
+* All devices with less than 7 MB of useable firmware space
```
Comment (mweinelt) [inline docs/dev/hardware.rst]:
```suggestion
* All devices with less than 7 MB of usable firmware space
```

File: docs/features/private-wlan.rst [hunk 20]
```diff
@@ -4,6 +4,14 @@ Private WLAN
 It is possible to set up a private WLAN that bridges the WAN port and is separated from the mesh network.
 Please note that you should not enable ``mesh_on_wan`` simultaneously.
 
+The private WLAN is encrypted via WPA2 by default. On devices with enough flash and a supported radio,
```
Comment (mweinelt) [inline docs/features/private-wlan.rst]:
```suggestion
The private WLAN is encrypted through WPA2 by default. On devices with enough flash and a supported radio,
```

File: docs/features/private-wlan.rst [hunk 21]
```diff
@@ -4,6 +4,14 @@ Private WLAN
...
+WPA3 or WPA2/WPA3 mixed-mode can be used instead of WPA2. For this to work, the ``wireless-encryption-wpa3``
+feature has to be added to the list of your supported features.
```
Comment (mweinelt) [inline docs/features/private-wlan.rst]: You are talking about `GLUON_FEATURES` here? If so add the variable name explicitly.

File: docs/features/private-wlan.rst [hunk 22]
```diff
@@ -4,6 +4,14 @@ Private WLAN
...
+
+Additionally, IEEE 802.11w management frame protection should be enforced for WPA2/WPA3, however this
+can lead to connectivity problems for older clients. In this case, management frame protection can be
+set to optional or disabled completely using advanced settings.
```
Comment (mweinelt) [inline docs/features/private-wlan.rst]: can be configured to be optional or disabled in advanced settings tab.

File: docs/site-example/site.conf [hunk 23]
```diff
@@ -42,10 +42,14 @@
     -- Wireless channel.
     channel = 1,
 
-    -- ESSID used for client network.
+    -- ESSIDs used for client network.
     ap = {
-      ssid = 'alpha-centauri.freifunk.net',
+      -- ssid = 'alpha-centauri.freifunk.net', (optional - SSID for open client network)
```
Comment (mweinelt) [inline docs/site-example/site.conf]: Comment above the commented out config line please.

File: docs/features/private-wlan.rst [hunk 24]
```diff
@@ -4,6 +4,14 @@ Private WLAN
 It is possible to set up a private WLAN that bridges the WAN port and is separated from the mesh network.
 Please note that you should not enable ``mesh_on_wan`` simultaneously.
 
+The private WLAN is encrypted through WPA2 by default. On devices with enough flash and a supported radio,
+WPA3 or WPA2/WPA3 mixed-mode can be used instead of WPA2. For this to work, the ``wireless-encryption-wpa3``
+feature has to be added to ``GLUON_FEATURES``.
+
+Additionally, IEEE 802.11w management frame protection should be enforced for WPA2/WPA3, however this
+can lead to connectivity problems for older clients. In this case, management frame protection can be
```
Comment (mweinelt) [inline docs/features/private-wlan.rst]:
```suggestion
can lead to connectivity problems for older clients. Therefore management frame protection can be
```

File: docs/features/private-wlan.rst [hunk 25]
```diff
@@ -4,6 +4,14 @@ Private WLAN
...
+configured to be optional or disabled in the advanced settings tab.
```
Comment (mweinelt) [inline docs/features/private-wlan.rst]:
```suggestion
made be optional or completely disabled in the advanced settings tab.
```

File: docs/user/site.rst [hunk 26]
```diff
@@ -130,8 +130,14 @@ wifi24 \: optional
     This will only affect new installations.
     Upgrades will not change the disabled state.
 
-    ``ap`` requires a single parameter, a string, named ``ssid`` which sets the
-    interface's ESSID. This is the WiFi the clients connect to.
+    ``ap`` holds the client network configuration.
+    To create an unencrypted client network, a string named ``ssid`` which sets the
+    interface's ESSID is required. This is the WiFi the clients connect to.
```
Comment (mweinelt) [inline docs/user/site.rst]:
```suggestion
    interface's ESSID is required. This is the wireless network clients connect to.
```

File: docs/user/site.rst [hunk 27]
```diff
@@ -130,8 +130,14 @@ wifi24 \: optional
...
+    For an OWE secured network, the ``owe_ssid`` string has to be set. It sets the
+    SSID for the OWE network compatible clients can connect to.
```
Comment (mweinelt) [inline docs/user/site.rst]:
```suggestion
    SSID for the opportunistically encrypted wireless network, to which compatible clients can connect to.
```

File: docs/user/site.rst [hunk 28]
```diff
@@ -527,14 +535,32 @@ GLUON_DEPRECATED
...
 GLUON_FEATURES
-    Defines a list of features to include. The feature list is used to generate
-    the default package set.
+    Defines a list of features to include. Depending on the device, the feature list
+    defined from this value is combined with the feature list for either the standard
+    or the tiny device-class. The resulting feature list is used to generate the default
+    package set.
+
+GLUON_FEATURES_standard
+    Defines a list of additional packages to include or exclude for devices of
```
Comment (mweinelt) [inline docs/user/site.rst]:
```suggestion
    Defines a list of additional features to include or exclude for devices of
```

File: package/features [hunk 29]
```diff
@@ -32,3 +32,6 @@ packages 'mesh-batman-adv-15' \
 
 packages 'mesh-babel' \
 	'gluon-radvd'
+
+packages '!wireless-encryption-wpa3' \
+	'gluon-wireless-encryption-wpa2'
```
Comment (mweinelt) [inline package/features]: How does this work? If I don't include the private wifi feature, will I have the `gluon-wireless-encryption-wpa2` package included by default?

Comment (blocktrron) [reply package/features]: Exactly.

**Comment by txt-file** [comment]
Which hardware/software is capable of WPA3-OWE today?
iOS 13.4 showed me some login screen instead of connecting. Tested with a sk-wb8 (ramips-mt76x8) running OpenWrt master.

**Comment by mweinelt** [comment]
Tested OWE (in Transition Mode) today.

- Android 10 on Pixel 3a 
  - Enhanced Open :heavy_check_mark: 
  - https://shells.darmstadt.ccc.de/~hexa/owe/pixel3a-android10/
- iOS 13.3.1 on iPhone XS 
  - Falls back to Open :white_check_mark: 
  - https://shells.darmstadt.ccc.de/~hexa/owe/iphone-xs-ios-13.3.1/

**Comment by blocktrron** [comment]
I can confirm this is working with and without transition mode on my Nokia 6.1 with Android 10.

On Android 9, the same device had trouble connecting to a transition SSID but worked fine with the dedicated OWE SSID. As OWE was not officially supported in Android 9, i assume this is due to an incorrect implementation on the phones side.

There's also this PR enabling transition mode support for NetworkManager:
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/442 

NetworkManager already supports dedicated OWE SSIDs since
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/345#note_328850 but AFAIK no distribution currently ships a release with this feature.

File: docs/user/site.rst (hunk 5 above)
Comment (blocktrron) [reply docs/user/site.rst]: Altered to allow any combination.

File: package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua [hunk 30]
```diff
@@ -61,21 +61,45 @@ uci:foreach('wireless', 'wifi-device', function(config)
...
+		local out = {}
+		for _, interface in ipairs(interfaces) do
+			if uci:get('wireless', interface .. '_' .. radio) then
+				table.insert(out, interface)
+			end
+		end
+		return out
+	end
+
+	local function are_interfaces_active(interfaces)
```
Comment (mweinelt) [inline package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua]:
```suggestion
	local function filter_active_interfaces(interfaces)
```

**Comment by mweinelt** [comment]
How are OWE clients exposed via respondd?

File: package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless [hunk 31]
```diff
@@ -42,6 +38,84 @@ util.foreach_radio(uci, function(radio, index, config)
...
+	uci:section('wireless', 'wifi-iface', name, {
+		device = radio_name,
+		network = 'client',
+		mode = 'ap',
+		ssid = ap.owe_ssid(),
+		macaddr = macaddr,
+		ifname = suffix and 'owe' .. suffix,
+		disabled = disabled or false,
+		encryption = 'owe',
+		mfp = 2,
```
Comment (blocktrron) [inline package/gluon-client-bridge/luasrc/lib/gluon/upgrade/320-gluon-client-bridge-wireless]: Has to be `ieee80211` instead of `mfp`

**Comment by blocktrron** [comment]
OWE clients are now exposed via respondd.

We've rolled out this patchset ~1 week ago to parts of our network: https://darmstadt.freifunk.net/news/2020/03/26/firmware-wpa3.html
